corehq/apps/users/middleware.py

```diff
@@ -15,6 +16,9 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if 'org' in view_kwargs:
             request.org = view_kwargs['org']
         if request.user and request.user.is_authenticated:
+            if not request.user.is_active:
+                logout(request)
+                return self.get_response(request)
```

Comment:
I'm not sure this is actually what we want in the long run based on [docs](https://docs.djangoproject.com/en/4.2/topics/http/middleware/#middleware-order-and-layering), as it seems by calling `self.get_response(request)` we will continue to pass the request down the middleware chain. Since the user has been logged out, I imagine it hits a point where an authenticated session is required and returns a response then. This doesn't seem too risky, and the overall change is still an improvement so I don't consider this blocking, but would like to see this potentially addressed if needed in the future.